src/api/ingest/utils.py

```diff
@@ -699,149 +36,58 @@ def import_household(datafile, identifier, cleardata, tofile):
-    thechoices = get_ingest_project_choices()
-
-    ## check for matching lookups
 
-    lookuperrors = []
-    lookupfields = [
-        # ["mpa", "mpa"],
-        ["settlement", "settlement"],
-        ["primaryinterviewer", "enumerator"],
-        ["secondaryinterviewer", "enumerator"],
-        ["fieldcoordinator", "fieldcoordinator"],
-        ["surveyversionnumber", "surveyversion"],
-        ["primarylivelihood", "livelihood"],
-        ["secondarylivelihood", "livelihood"],
-        ["tertiarylivelihood", "livelihood"],
-        ["freqfishtime", "freqfishtime"],
-        ["freqeatfish", "freqfishtime"],
-        ["freqsalefish", "freqfishtime"],
-        ["percentproteinfish", "nonetoallscale"],
-        ["fsnotenough", "foodsecurity"],
-        ["fsdidnotlast", "foodsecurity"],
-        ["fsbalanceddiet", "foodsecurity"],
-        ["dataentryid", "enumerator"],
-        ["datacheckid", "enumerator"],
-        ["majorfishtechnique", "majorfishtech"],
-    ]
-    # print (thechoices["data__enumerator"])
-    # return
+    rowcount = 0
     for row in reader:
-        for thefld in lookupfields:
+        rowcount += 1
+        row = {key.strip(): value for key, value in row.items()}  # some field names have spaces at the end
+
+        for lookup_model in related_models:
+            rec = lookup_model[0]
             try:
-                lookupvalue = (
-                    thechoices["data__" + thefld[1]][str(row[thefld[0]])]
-                    if row[str(thefld[0])]
-                    else None
-                )
-            except (KeyError):
-                lookuperrors.append([row["householdid"], row[thefld[0]], thefld[0]])
+                lookup_instance = None
+                if row[rec]:
+                    lookup_instance = lookup_model[1].objects.get(pk=row[rec])
+            except lookup_model[1].DoesNotExist:
+                lookuperrors.append([rowcount, row[rec], rec])
                 continue
-    print("check complete")
+            row[rec] = lookup_instance
 
-    datafile.seek(0)
-    reader = csv.DictReader(datafile)
-    if tofile:
-        now = datetime.now()
-        current_time = now.strftime("%H_%M_%S")
-        f = open("household_" + current_time + ".txt", "a")
-
-    if len(lookuperrors):
-        display_errors(lookuperrors, tofile, identifier)
-    for row in reader:
-        print(row["householdid"])
+        # remove fields that are blank
+        cleaned_row = {k: v for k, v in row.items() if v}
 
-        for thefld in lookupfields:
-            row[thefld[0]] = (
-                thechoices["data__" + thefld[1]][row[thefld[0]]]
-                if row[thefld[0]]
-                else None
-            )
+        #convert date fields to YYYY-MM-DD if they are in MM-DD-YYYY
+        for fld_date in themodel._meta.get_fields():
+            if fld_date.__class__.__name__ == "DateField":
+                try:
+                    datetime.strptime(cleaned_row[fld_date.name], "%m/%d/%Y")
```

Comment:
At one point we were talking about separating the csv columns into month,date,year -- that would be preferable to relying on a particular string pattern.